model_tests.rb

```diff
@@ -858,6 +859,205 @@ def autosized_fields(model, obj_types_to_skip, missing_getters)
+      "seed_file": "in.osm",
+      # "steps": [],
+    }
+
+    File.write(osw, JSON.pretty_generate(osw_content))
+
+
+    # Save to above model into dir
+    m.save(in_osm, true)
+
+    # Run it
+    command = "\"#{$OpenstudioCli}\" run -w \"#{osw}\""
+    #command = "\"#{$OpenstudioCli}\" run --debug -w \"#{osw}\""
+
+    run_command(command, dir, 3600)
+  end
+
+  # Typically we do this:
+  # result_osw = postprocess_out_osw_and_copy(filename)
+  # Here we actually don't care about the OSW at all,
+  # just want to make sure that E+ completed successfully
+  # (though I don't see a reason it wouldn't)
+  fail "Cannot find file #{out_osw}" if !File.exists?(out_osw)
+
+  result_osw = nil
+  File.open(out_osw, 'r') do |f|
+    result_osw = JSON::parse(f.read, :symbolize_names=>true)
+  end
+
+  assert_equal("Success", result_osw[:completed_status])
+
+  # Load the model
+  versionTranslator = OpenStudio::OSVersion::VersionTranslator.new
+
+  model = versionTranslator.loadModel(in_osm)
+  if model.empty?
+    assert(model.is_initialized, "Could not load the resulting model, #{in_osm}")
+  end
+  model = model.get
+
+  # Load and attach the sql file to the model
+  sql_path = OpenStudio::Path.new(sql_path)
+  if OpenStudio.exists(sql_path)
+    sql = OpenStudio::SqlFile.new(sql_path)
+    # Check to make sure the sql file is readable,
+    # which won't be true if EnergyPlus crashed during simulation.
+    unless sql.connectionOpen
+      OpenStudio.logFree(OpenStudio::Error, 'openstudio.model.Model', "The run failed, cannot create model.  Look at the eplusout.err file in #{File.dirname(sql_path.to_s)} to see the cause.")
+      return false
+    end
+    # Attach the sql file from the run to the model
+    model.setSqlFile(sql)
+  else
+    OpenStudio.logFree(OpenStudio::Error, 'openstudio.model.Model', "Results for the run couldn't be found here: #{sql_path}.")
+    return false
+  end
+
+  # Assert that the sizing run succeeded
+  assert_equal("Success", result_osw[:completed_status])
+
+  # Now do actual stuff!
+
+  # Get the daily series
+  ts = sql.timeSeries(sql.availableEnvPeriods[0], "Daily", "Site Outdoor Air Drybulb Temperature", "Environment")
+  assert(ts.is_initialized, "Timeseries isn't initialized")
+  ts = ts.get
+
+
+  # End date minus start date, + 1 because we count the start AND the end day
+  n_days = (d_end - d_start).totalDays + 1
+  assert_equal(n_days, ts.values.size, "Bad number of days!")
+
+  # E+ reports with an end convention, meaning that for a daily timeseries
+  # each entry is labeled as the next day
+  # eg: if you start on 2009-01-01, eplus reports as 2009-01-02 00:00
+  sql_start_dt = ts.firstReportDateTime
+  sql_end_dt = ts.dateTimes[-1]
+
+  # Easy check first, check that the year was correctly sent to E+
+  assert_equal(start_year, sql_start_dt.date.year, "Start year doesn't match")
+
+  # Remove 1 day so we can compare with the above
+  sql_start_date = (sql_start_dt - OpenStudio::Time.new(1)).date
```

Comment:
Not quite sure I follow this. For the simple case of 1/1/2009-12/31/2009 what are sql_start_date and sql_end_date?